tests/compile/tsick.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Lightweight conversion from tsc to Closure Compiler.
+ */
+'use strict';
+
+const fs = require('fs');
+
+const DIR = '../../build/src/core/';
+
+function spider(dir) {
+  const data = fs.readdirSync(dir, {withFileTypes: true});
+  for (const datum of data) {
+    if (datum.isDirectory()) {
+      spider(dir + datum.name + '/')
+    } else if (datum.name.endsWith('.js')) {
+      rewriteFile(dir + datum.name);
+    }
+  }
+}
+
+function rewriteFile(path) {
+  const oldCode = fs.readFileSync(path, 'utf8');
+  const newCode = rewriteEnum(oldCode);
+  if (newCode !== oldCode) {
+    console.log('Rewrote: ' + path);
+    fs.writeFileSync(path, newCode);
+  }
+}
+
+function rewriteEnum(code) {
+  while (true) {
+    // Extract the entire enum structure.
+    const m = code.match(/\s+\(function \((\w+)\) \{\n[^\}]*\}\)\(\1 [^)]+\1 = \{\}\)\);/);
+    if (!m) {
+      break;
+    }
+    // m[0] looks like a bunch of lines in one of these two formats:
+    //   ScopeType["BLOCK"] = "block";
+    //   KeyCodes[KeyCodes["TAB"] = 9] = "TAB";
+    // We need to unquote them to look like one of these two formats:
+    //   ScopeType.BLOCK = "block";
+    //   KeyCodes[KeyCodes.TAB = 9] = "TAB";
+    const oldSnippet = m[0];
+    let newSnippet = oldSnippet.replace(/\["(\w+)"\]/g, '.$1');
+    // Add a comment so we don't keep trying to modify this enum.
+    newSnippet = newSnippet.replace(') {', ') {  // Converted by tsick.');
```

Comment:
Judicious use of the [the stick`y` flag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky) should obviate the need for this (but do make sure you're not creating a fresh RegExp object, and thereby resetting `.lastIndex`, for each iteration!)

Reply:
The `y` flag is giving me mixed messages.  Here it works as expected:
```
const ry = /m/y;
'mmm'.match(ry);
'mmm'.match(ry);
```
But here it does not, returning null on the second call:
```
const ry = /m/y;
'm_m_m'.match(ry);
'm_m_m'.match(ry);
```